src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -848,11 +862,12 @@ ReactDOMComponent.Mixin = {
         CONTENT_TYPES[typeof props.children] ? props.children : null;
       var childrenToUse = contentToUse != null ? null : props.children;
       if (contentToUse != null) {
-        // TODO: Validate that text is allowed as a child of this node
-        if (__DEV__) {
-          setAndValidateContentChildDev.call(this, contentToUse);
+        if (shouldSetNodeTextContent(lazyTree, props, contentToUse)) {
```

Comment:
Let's just do `contentToUse !== ''` here (and move the comment you have above down here to explain). No need to check for textarea/placeholder.